### PR TITLE
numfmt: replace getopts with clap

### PR DESCRIPTION
**Pull request by drocco007**
Hi! This small contribution converts argument parsing in the numfmt utility from getopts to clap.

The test suite passes and clippy and rustfmt are happy.

Please let me know if I can improve this PR.

Thanks!

### CODE REVIEW

File: src/uu/numfmt/src/numfmt.rs [hunk 1]
```diff
@@ -279,83 +303,57 @@ fn handle_stdin(options: NumfmtOptions) -> Result<()> {
...
-    }
+    let usage = get_usage();
+
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about(ABOUT)
+        .usage(&usage[..])
+        .after_help(LONG_HELP)
+        .arg(
+            Arg::with_name("from")
```
Comment (sylvestre) [inline src/uu/numfmt/src/numfmt.rs:314]: For consistency, could you follow what is done elsewhere with
OPT_FROM 
and sort by alphabetical order? Example:
https://github.com/uutils/coreutils/blob/master/src/uu/mktemp/src/mktemp.rs#L30-L35
Thanks

**Review by sylvestre — COMMENTED** [review]
Looks great, just a small change!

File: src/uu/numfmt/src/numfmt.rs (hunk 1 above)
Comment (sylvestre) [reply src/uu/numfmt/src/numfmt.rs:314]: Or better, do like https://github.com/uutils/coreutils/pull/1712/files
(mod options { })

Comment (drocco007) [reply src/uu/numfmt/src/numfmt.rs]: Updated using the `mod options` style.

Comment (sylvestre) [reply src/uu/numfmt/src/numfmt.rs]: @FelipeLema how do you feel? 
(sorry I said conflicting things, I am still a bit unsure :)

Comment (FelipeLema) [reply src/uu/numfmt/src/numfmt.rs]: ok, so this is about the following story (my personal view)

Say I'm reading some code one day and I stand on something like `option::extra_args::cleanup`.

I can infer certain things just when looking at this:
- `cleanup` is a (command line) option
- it's one of several
- they have some kind of (hierarchical) order, so I can narrow my scope if I'm dealing with this.

Now, if I want to look into related code, there's a couple of things I can do

When doing "go to definition" using my IDE (with either rust-analyzer or some ad-hoc tagging system) I can jump to both the definition _and the other related options_ because they're all in the same (sub-)module.

Why can't I do this with `OPT_`? Because relationship between options is either implicit or has to be _semantically_ understood. Either way, it's an overload that the dev has to pay.

Yes, packing options into modules make an almost insignificant difference each, but people (me) will thank you for it because I *know* that extra effort sums up when dealing with big code bases.

TL;DR: Classifying options into modules improves maintainability according to my experience. Specially in large code bases.

Comment (FelipeLema) [reply src/uu/numfmt/src/numfmt.rs]: (I also think that packing options in modules can be done in separate features / commits ... so go for it!)

**Comment by chadbrewbaker** [comment]
Could we add a test case that stresses the memory, even if it is turned off by default in the test suite? Like to know the opt/clap resource difference and clap's failure modes.

**Comment by sylvestre** [comment]
@chadbrewbaker this PR with about getopts => clap migration. Please open a separate issue for your request! thanks!

File: src/uu/numfmt/src/numfmt.rs (hunk 1 above)
Comment (sylvestre) [reply src/uu/numfmt/src/numfmt.rs]: agreed, many thanks for your arguments :)

**Comment by chadbrewbaker** [comment]
I had a more general test rig in mind that ran everything under /usr/bin/time -v  so we have time/memory regressions, and a ulimit script to elicit out of memory failure modes.  Let me think about it.

File: src/uu/numfmt/src/numfmt.rs (hunk 1 above)
Comment (drocco007) [reply src/uu/numfmt/src/numfmt.rs]: @FelipeLema thank you for explaining your thoughtful reasoning
